internal/drain-config/drainConfig.go

```diff
@@ -0,0 +1,179 @@
+package drain_config
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	migrator "github.com/rudderlabs/rudder-server/services/sql-migrator"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+const (
+	defaultPollFrequency      = 5
+	defaultPollFrequencyUnits = time.Second
+
+	defaultCleanupFrequency      = 24
+	defaultCleanupFrequencyUnits = time.Hour
+
+	defaultMaxAge      = 24
+	defaultMaxAgeUnits = time.Hour
+
+	// drain configurations
+
+	jobRunID       = "drain.jobRunIDs"
+	configJobRunID = "RSources.toAbortJobRunIDs"
+)
+
+type drainConfigManager struct {
+	log  logger.Logger
+	conf *config.Config
+	db   *sql.DB
+}
+
+func NewDrainConfigManager(conf *config.Config, log logger.Logger) (*drainConfigManager, error) {
+	db, err := setupDBConn(misc.GetConnectionString(conf))
+	if err != nil {
+		log.Errorw("db setup", "error", err)
+		return nil, fmt.Errorf("db setup: %v", err)
+	}
+	if err := migrate(db); err != nil {
+		log.Errorw("db migrations", "error", err)
+		return nil, fmt.Errorf("db migrations: %v", err)
+	}
+	return &drainConfigManager{
+		log:  log,
+		conf: conf,
+		db:   db,
+	}, nil
+}
+
+func (d *drainConfigManager) CleanupRoutine(ctx context.Context) error {
+	for {
+		maxAgeInMinutes := d.conf.GetDuration("drain.age", defaultMaxAge, defaultMaxAgeUnits) / time.Minute
```

Comment:
What is the intention here? Why don't we calculate the expiration time instead?
e.g.

```go
expiryTime := time.Now().Add(-d.conf.GetDuration("drain.age", defaultMaxAge, defaultMaxAgeUnits))
```